### PR TITLE
fix(app-switcher-close): close exp panel on menu close, detect changes

### PR DESCRIPTION
**Pull request by thomaslennan**
When App Switcher mat-menu was closing, corresponding expansion panel child might be left in open state. Forced close of expansion panel on mat-menu close. Also added detect changes before scroll attempt to make sure that the component height was computed correctly.

### CODE REVIEW

**Review by JulieMarie — CHANGES_REQUESTED** [review]
Two things:
1. Replace `http://teradata.com` with `https://www.teradata.com`.  I know it's nothing you did, but just an overall search/replace with make testing internally easier.
2. One oddity with the drop down/scroll up caret.  See animated gif and ask me if you have any questions.
vantage-ui-platform

**Review by JulieMarie — COMMENTED** [review]
It's weird that the dropdown is chunky and sluggish (at least for me) the first time I click on it to dropdown, but then it's fine after that!

I would like to see `npm audit` fixed, but that's not a requirement.

File: src/lib/app-switcher/app-switcher-menu/app-switcher-menu.component.ts [hunk 1]
```diff
@@ -13,17 +15,21 @@ export class VantageAppSwitcherMenuComponent {
...
     event.preventDefault();
     event.stopPropagation();
   }
 
   scrollToBottom(): void {
+    this._changeDetectorRef.detectChanges();
     setTimeout(() => {
       const elem: HTMLElement = this.elRef.nativeElement.querySelector('.td-menu-content');
       elem.scrollTop = elem.scrollHeight;
+      this._changeDetectorRef.detectChanges();
```
Comment (emoralesb05) [inline src/lib/app-switcher/app-switcher-menu/app-switcher-menu.component.ts:32]: Does this not work with `markForChange`? 

detectChanges means we did something wrong if we have to use it for the most part.

File: src/lib/app-switcher/app-switcher.component.ts [hunk 2]
```diff
@@ -12,4 +13,12 @@ export class VantageAppSwitcherComponent {
   @Input() products: IVantageAppSwitcherItem[];
   @Input() otherProducts: IVantageAppSwitcherItem[];
   @Input() exploreMoreLink: string;
+  @ViewChild('appSwitcherMenu', { static: true }) appSwitcherMenu: VantageAppSwitcherMenuComponent;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+
+  menuClosed(): void {
+    this.appSwitcherMenu.expansionPanel.close();
+    this._changeDetectorRef.detectChanges();
```
Comment (emoralesb05) [inline src/lib/app-switcher/app-switcher.component.ts:22]: Same as the previous comment.

**Review by emoralesb05 — CHANGES_REQUESTED** [review]
Left a few comments

**Review by emoralesb05 — APPROVED** [review]

**Review by JulieMarie — APPROVED** [review]
LGTM

**Pull request merged**